subprojects/testfx-core/src/test/java/org/testfx/util/DebugUtilsTest.java

```diff
@@ -0,0 +1,296 @@
+/*
+ * Copyright 2013-2014 SmartBear Software
+ * Copyright 2014-2017 The TestFX Contributors
+ *
+ * Licensed under the EUPL, Version 1.1 or - as soon they will be approved by the
+ * European Commission - subsequent versions of the EUPL (the "Licence"); You may
+ * not use this work except in compliance with the Licence.
+ *
+ * You may obtain a copy of the Licence at:
+ * http://ec.europa.eu/idabc/eupl.html
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed
+ * under the Licence is distributed on an "AS IS" basis, WITHOUT WARRANTIES OR
+ * CONDITIONS OF ANY KIND, either express or implied. See the Licence for the
+ * specific language governing permissions and limitations under the Licence.
+ */
+package org.testfx.util;
+
+import java.util.concurrent.TimeoutException;
+import java.util.function.Function;
+import java.util.function.UnaryOperator;
+
+import javafx.beans.property.SimpleBooleanProperty;
+import javafx.scene.Scene;
+import javafx.scene.input.KeyCode;
+import javafx.scene.input.MouseButton;
+import javafx.scene.layout.Region;
+import javafx.stage.Stage;
+
+import org.hamcrest.BaseMatcher;
+import org.hamcrest.Description;
+import org.hamcrest.Matcher;
+import org.junit.After;
+import org.junit.Rule;
+import org.junit.Test;
+import org.testfx.TestFXRule;
+import org.testfx.api.FxAssert;
+import org.testfx.api.FxToolkit;
+import org.testfx.cases.TestCaseBase;
+import org.testfx.service.support.FiredEvents;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.testfx.util.DebugUtils.compose;
+import static org.testfx.util.DebugUtils.insertHeader;
+import static org.testfx.util.DebugUtils.runCode;
+import static org.testfx.util.DebugUtils.showFiredEvents;
+import static org.testfx.util.DebugUtils.showKeysPressedAtTestFailure;
+import static org.testfx.util.DebugUtils.showMouseButtonsPressedAtTestFailure;
+
+public class DebugUtilsTest extends TestCaseBase {
+
+    private static final String TAB = " ";
+
+    private static final MouseButton MOUSE_BUTTON = MouseButton.PRIMARY;
+    private static final KeyCode KEY = KeyCode.A;
```

Comment:
I don't think that these fields add anything useful. I understand if the intention is to make it clear that it can be any mouse button or key, but I think it complicates reading the test.